basic_machines/distributor.lua

```diff
@@ -240,17 +229,24 @@ local function tubelib2_on_update2(pos, outdir, tlib2, node)
 	end
 end
 
-local function push_item(pos, filter, itemstack, num_items, nvm)
+local function shuffle(list)
+	for i = #list, 2, -1 do
+		local j = math.random(i)
+		list[i], list[j] = list[j], list[i]
+	end
+	return list
+end
+
+local function push_item(pos, base_filter, itemstack, num_items, nvm)
+	local filter = shuffle(table.copy(base_filter))
 	local idx = 1
 	local num_pushed = 0
 	local num_ports = #filter
```

Comment:
#filter can become zero. It was the cause for several server crashes. This was the reason for line 247 which was added lately. 
```
num_ports = techage.in_range(num_ports, 1, 4)
```

This means, the line
```
local filter = shuffle(table.copy(base_filter))
```
but also
```
local amount = math.floor(math.max((num_items + 1) / num_ports, 1))
```
will crash if this happens again.